rackcity/views/change_plan_views.py

```diff
@@ -254,3 +267,99 @@ def change_plan_detail(request, id):
         },
         status=HTTPStatus.OK
     )
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def change_plan_execute(request):
+    """
+    Execute all changes associated with a change plan.
+    """
+    data = JSONParser().parse(request)
+    if 'id' not in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.DELETE_ERROR.value + GenericFailure.INTERNAL.value,
+                "errors": "Must include 'id' when deleting a Change Plan"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    id = data['id']
+    try:
+        change_plan = ChangePlan.objects.get(id=id)
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.ERROR.value +
+                    "Change plan" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing change plan with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    if request.user != change_plan.owner:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.ERROR.value +
+                    "You do not have access to execute this change plan.",
+                "errors":
+                    "User " + request.user.username +
+                    "does not own change plan with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
```

Comment:
good to have this check in place, but generally this should never happen right? bc only a user can see their own chagne plans

Reply:
correct